lib/pbench/cli/agent/commands/results/push.py

```diff
@@ -24,10 +25,30 @@ def execute(self) -> int:
             self.config,
             self.logger,
         )
-        crt.copy_result_tb(
+        res = crt.copy_result_tb(
             self.context.token, self.context.access, self.context.metadata
         )
-        return 0
+
+        # success
+        if res.status_code == 201:
+            return 0
+
+        try:
+            msg = res.json()["message"]
+        except requests.exceptions.JSONDecodeError:
+            msg = res.text if res.text else res.reason
+
+        # dup
+        if res.status_code == 200:
+            click.echo(msg, err=True)
+            return 0
+
+        if not res.ok:
+            click.echo(
+                f"HTTP Error status: {res.status_code}, message: {msg}",
+                err=True,
+            )
+            return 1
 
 
 @sort_click_command_parameters
```

Comment:
If the test at line 46 fails, then this function will return `None`, which is undesirable.  (I don't _expect_ [that test](https://requests.readthedocs.io/en/latest/api/#requests.Response.ok) to fail, but, in the unlikely event that we get a status less than `400` which isn't `200` or `201` (such as a `100` or a `30x`, which might have been sent by a proxy and for some reason returned by the `requests` package), it would be good to report that rather than presuming it to be "OK" or otherwise confusing our caller.)

So, I recommend leaving the block at line 47 to be executed unconditionally, in the event that control reaches it.